spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java

```diff
@@ -600,7 +603,16 @@ public void testBatchRecords() throws Exception {
 		assertThat(this.config.listen12Message.getPayload()).isInstanceOf(List.class);
 		List<?> errorPayload = (List<?>) this.config.listen12Message.getPayload();
 		assertThat(errorPayload.size()).isGreaterThanOrEqualTo(1);
+	}
+
+	@Test
+	public void testBatchInterceptor() throws Exception {
+		template.send("annotated43", null, "foo");
+		assertThat(this.config.batchAfterRecordsProcessedLatch.await(30, TimeUnit.SECONDS)).isTrue();
```

Comment:
Because I can't wait on a latch inside `@KafkaListener`, it's too early and the test might run next assertion before the interceptor methods are called. Current approach doesn't work because `batchAfterRecordsProcessedLatch` is called too early, in the previous `pollAndInvoke()` iteration before the actual test sends the message. I need to improve it.